x-pack/plugins/cases/public/containers/use_get_action_license.tsx

```diff
@@ -5,81 +5,34 @@
  * 2.0.
  */
 
-import { useCallback, useEffect, useState, useRef } from 'react';
-
+import { useQuery } from 'react-query';
 import { useToasts } from '../common/lib/kibana';
 import { getActionLicense } from './api';
 import * as i18n from './translations';
-import { ActionLicense } from './types';
 import { ConnectorTypes } from '../../common/api';
-
-export interface ActionLicenseState {
-  actionLicense: ActionLicense | null;
-  isLoading: boolean;
-  isError: boolean;
-}
-
-export const initialData: ActionLicenseState = {
-  actionLicense: null,
-  isLoading: true,
-  isError: false,
-};
+import { CASE_LIST_CACHE_KEY } from './constants';
+import { ServerError } from '../types';
 
 const MINIMUM_LICENSE_REQUIRED_CONNECTOR = ConnectorTypes.jira;
 
-export const useGetActionLicense = (): ActionLicenseState => {
-  const [actionLicenseState, setActionLicensesState] = useState<ActionLicenseState>(initialData);
+export const useGetActionLicense = () => {
   const toasts = useToasts();
-  const isCancelledRef = useRef(false);
-  const abortCtrlRef = useRef(new AbortController());
-
-  const fetchActionLicense = useCallback(async () => {
-    try {
-      isCancelledRef.current = false;
-      abortCtrlRef.current.abort();
-      abortCtrlRef.current = new AbortController();
-      setActionLicensesState({
-        ...initialData,
-        isLoading: true,
-      });
-
-      const response = await getActionLicense(abortCtrlRef.current.signal);
-
-      if (!isCancelledRef.current) {
-        setActionLicensesState({
-          actionLicense: response.find((l) => l.id === MINIMUM_LICENSE_REQUIRED_CONNECTOR) ?? null,
-          isLoading: false,
-          isError: false,
-        });
-      }
-    } catch (error) {
-      if (!isCancelledRef.current) {
+  return useQuery(
+    [CASE_LIST_CACHE_KEY, 'action-license'],
```

Comment:
I am wondering if we should tie up the `action-license` with the case list page. What if we want to get the action licenses to another page (configuration page for example) and we want to invalidate the query?

Reply:
In that case then we should give it its own cache key.